### ISSUE TITLE
Help Required: Unable to validate LocalDateTime (`YYYY-MM-DDTHH:mm:ss`) for payload in POST endpoint

### ISSUE THREAD

**Issue by bh4r4th** [devopsfaith/krakend-ce#438]
I am trying to create a POST endpoint which accepts payload shape:

```
{

  "2022-03-17T10:30:00" :  [
    {
      "pId": "some-str-01",
      "val": "abc"
    }
  ],

  "2022-03-17T20:30" :  [
    { "pId": "some-str-02",  "val": "abc" },
    { "pId": "some-str-03",  "val": 123 },
    { "pId": "some-str-04",  "val": 123.12 },
    { "pId": "some-str-05",  "val":  false },
    { "pId": "some-str-06",  "val": {} },
    { "pId": "some-str-07",  "val": [] },
  ],

}
```
Which is an object of `LocalDateTime` as `keys` and `List of Objects` as `values`.

**What I have tried:**
```
{
      "endpoint": "/v1/category/{category_id}/history",
      "method": "POST",
      "extra_config": {
        "github.com/devopsfaith/krakend-ratelimit/juju/router": {
          "maxRate": 20,
          "clientMaxRate": 10,
          "strategy": "ip"
        },
        "validation/json-schema": {
          "type": "object",
          "propertyNames": {
            "type": "string",
            "format": "date-time"
          },
          "additionalProperties": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "pId": { "type": "string" },
                "val": { "type": ["number", "boolean", "string", "array", "object"] }
              },
              "required": ["pId", "val"]
            }
          }
        }
      },
      "backend": [
        {
          "url_pattern": "/v1/category/{category_id}/history",
          "method": "POST",
          "host": [
            "{{ env "PRODUCT_API_HOST"}}"
          ],
          "extra_config": {
            "backend/http": {
              "return_error_details": "backend_product_api"
            }
          }
        }
      ]
 }
```

The `validation/json-schema` works fine for the ISO DateTime format with Zone offsets (OffsetDateTime format in Java).

For example: `2022-03-17T10:30:00+11:00` or `2022-03-16T23:30:00Z`

Can someone help me how can I apply a validation for DateTime (LocalDateTime format in Java). As `json-schema` only supports formats:
1. datetime  (which is OffsetDateTime  `YYYY-MM-DDTHH:mm:ssZ` or `YYYY-MM-DDTHH:mm:ss+HH:mm` or `YYYY-MM-DDTHH:mm:ss-HH:mm`)
2. date (which is LoalDate ` YYYY-MM-DD`)
3. time (which is localtime `HH:mm:ss`)
4. duration (which is duration in days.)

**What I am trying now is:**
Use of `regular expression` pattern for LocalDateTime like:

```
"propertyNames": {
            "type": "string",
            "pattern": "regex-to-validate-localdatetime-is-valid"
 }
```

But this is gross, as the pattern is huge when I include year range (1900-2999) with leap year and month with 30 and 31 days validation.

Can someone shed some light in doing this in a better way.

**Comment by alombarte** [comment]
Hi @bh4r4th ,

You would probably get better answers on the JSON schema repository. The question is not related to KrakenD itself, but to the usage of json schema.

**Comment by bh4r4th** [comment]
I reached out to the `json-schema` admin on slack. They're very responsive and mentioned that as ISO isn't an open standard they didn't feel it was OK to reference its use. 

In this particular scenario, I used a regex pattern to match the `propertyNames` which vaguely matches the format as `LocalDateTime`. If the timestamp is an invalid one, then my Kotlin (Java) API should handle it and return an error response. 

In KrakenD, I am doing a pass through for error response. This should solve my above mentioned problem :)
